src/storage-client/src/source/persist_source.rs

```diff
@@ -143,9 +154,51 @@ where
         Arc::new(metadata.relation_desc),
         Arc::new(UnitSchema),
         move |stats| {
-            mfp_pushdown.as_ref().map_or(true, |x| {
-                x.should_fetch(&PersistSourceDataStatsImpl { desc: &desc, stats })
-            })
+            if let Some(plan) = &filter_plan {
```

Comment:
nit: feels like this has passed the complexity threshold for being inlined in the closure definition. can we move the whole block out into a freestanding function?

bonus super nit: let's early return on the `if let Some(plan) = &filter_plan` by turning it in to a match with one case returning

(I think we actually want to move the bulk of this down into mz_expr somewhere, but that should be left for a later PR)